RealmSwift/SwiftUI.swift

```diff
@@ -0,0 +1,599 @@
+////////////////////////////////////////////////////////////////////////////
+ //
+ // Copyright 2021 Realm Inc.
+ //
+ // Licensed under the Apache License, Version 2.0 (the "License");
+ // you may not use this file except in compliance with the License.
+ // You may obtain a copy of the License at
+ //
+ // http://www.apache.org/licenses/LICENSE-2.0
+ //
+ // Unless required by applicable law or agreed to in writing, software
+ // distributed under the License is distributed on an "AS IS" BASIS,
+ // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ // See the License for the specific language governing permissions and
+ // limitations under the License.
+ //
+ ////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+
+#if canImport(SwiftUI) && canImport(Combine) && swift(>=5.3.1) && (REALM_HAVE_COMBINE || !SWIFT_PACKAGE)
+import SwiftUI
+import Combine
+import Realm
+import Realm.Private
+
+private func safeWrite<Value>(_ value: Value, _ block: () -> Void) where Value: ThreadConfined {
+    var didStartWrite = false
+    if value.realm?.isInWriteTransaction == false {
+        didStartWrite = true
+        value.realm?.beginWrite()
+    }
+    block()
+    if didStartWrite {
+        try! value.realm?.commitWrite()
+    }
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+private func createBinding<T: ThreadConfined, V>(_ value: T,
+                                                 forKeyPath keyPath: ReferenceWritableKeyPath<T, V>) -> Binding<V> {
+    if value.isFrozen {
+        throwRealmException("Should not bind frozen value")
```

Comment:
This appears to be something that's supposed to be unreachable. If that's correct, it should be a `precondition()` to make that clear, while if it's something the user actually can hit then it should be a bit more descriptive.